examples/graffiti.html

```diff
@@ -0,0 +1,433 @@
+  // the physical camera on the device.
+  camera = new THREE.ARPerspectiveCamera(vrDisplay, 60, window.innerWidth / window.innerHeight, 0.01, 100);
+
+  // VRControls is a utility from three.js that applies the device's
+  // orientation/position to the perspective camera, keeping our
+  // real world and virtual world in sync.
+  vrControls = new THREE.VRControls(camera);
+
+  // Bind our event handlers
+  window.addEventListener('resize', onWindowResize, false);
+
+  // Start drawing when the user touches the screen.
+  renderer.domElement.addEventListener('touchstart', function(event) {
+    drawing = true;
+  });
+
+  // Stop the current draw stroke when the user finishes the touch.
+  renderer.domElement.addEventListener('touchend', function(event) {
+    drawing = false;
+    stroke.length = 0;
+    strokeIndex += 1;
+  });
+
+  update();
+}
+
+// Get a point in front of the device to use as the drawing location.
+// Pass in a THREE.Vector3 to be populated with data to avoid creating garbage.
+// This strange function structure is a way of avoiding creating garbage while
+// generating the rotated forward vector.
+var getDrawPoint = (function() {
+  // Setup a basic forward vector (scaled down so it's not so far away).
+  var forward = new THREE.Vector3(0, 0, -1).multiplyScalar(STROKE_DISTANCE);
+
+  // Create a scratch vector for generating the rotated forward vector.
+  var rotatedForward = new THREE.Vector3();
+  return function(out) {
+    // Start with the camera position, which is equivalent to device pose.
+    out.copy(camera.position);
+
+    // Rotate the forward vector by the pose orientation.
+    rotatedForward.copy(forward);
+    rotatedForward.applyQuaternion(camera.quaternion);
+    out.add(rotatedForward);
+  }
+})();
+
+var strokeWidth = 0;
+var position = new THREE.Vector3();
+var previousPosition = new THREE.Vector3();
+var normal = new THREE.Vector3(0, 0, 1);
+var previousNormal = new THREE.Vector3(0, 0, 1);
+var velocity = new THREE.Vector3();
+
+function processDraw() {
+  if(stroke.length > MINIMUM_STROKE_POINTS && position.distanceTo(stroke[stroke.length - 1].position) < MINIMIM_POINT_DISTANCE) {
+    return;
+  }
+
+  // Add the draw point to the current stroke.
+  stroke.push(getStroke());
+
+  // Don't start drawing until there are enough points.
+  if (stroke.length < MINIMUM_STROKE_POINTS) {
+    return;
+  }
+
+  // Remove the old stroke from the scene so we can regenerate the stroke.
+  if (strokes[strokeIndex]) {
+    scene.remove(strokes[strokeIndex]);
+  }
+
+  strokes[strokeIndex] = generateStroke(stroke);
+  scene.add(strokes[strokeIndex]);
+}
+
+function getStroke()
+{
+  return {
+    position: new THREE.Vector3(position.x, position.y, position.z),
+    normal: new THREE.Vector3(normal.x, normal.y, normal.z),
+    velocity: new THREE.Vector3(velocity.x, velocity.y, velocity.z),
+    width: strokeWidth
+  };
+}
+
+function processDrawBrush() {
+  // Add the draw point to the current stroke.
```

Comment:
Not sure why but this code doesn't work. Brush/reticle is very glitchy. Something wrong here.

Reply:
I believe the newest commit should resolve the glitchy nature of the brush, it essentially hides it if there is no device motion.